login/login.c

```diff
@@ -383,23 +385,24 @@ int login_authenticate(glome_login_config_t* config, pam_handle_t* pamh,
   free(action);
   action = NULL;
 
-  const char* prefix = "";
-  if (config->url_prefix != NULL) {
-    prefix = config->url_prefix;
-  }
-  size_t message_len =
-      strlen(prompt_format) + strlen(prefix) - 2 + strlen(url) - 2 + 1;
+  const char* prompt = config->prompt;
+  size_t message_len = strlen(prompt) + strlen(challenge) + 1;
```

Comment:
Do we always have a prompt? I preferred the previous implementation with the helper variable `prefix`, which prevents me from thinking about the definedness of `strlen(NULL)`.